### PR TITLE
SQLite Rewrite

### PR DESCRIPTION
**Pull request by NiLSPACE**
Rewrites everything to use the latest API's and sqlite storage.

### CODE REVIEW

File: CommandHandlers.lua [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		a_Player:SendMessage("You are already logged in")
+		return true
+	end
+	
+	if ((a_Split[2] == nil) or (a_Split[3] == nil)) then
+		a_Player:SendMessage("Usage: /register <password> <confirmpassword>")
+		return true
+	end
+	
+	local res, Err = PlayerState:TryRegister(a_Split[2], a_Split[3])
```
Comment (madmaxoft) [inline CommandHandlers.lua]: I think the command handler should be responsible for checking that the two passwords match, and the `TryRegister()` function should only receive the single password.
If you ever decide that you don't want double-checking for the password, you can simply remove it in the handler. Or if you want to allow admin to register other people :)

File: Info.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		},
+		
+		['/register'] =
+		{
+			Permission = "login.register",
+			HelpString = "Register an account =",
+			Handler    = HandleRegisterCommand,
+		},
+	},
+}
```
Comment (madmaxoft) [inline Info.lua]: Missing newline

**Comment by madmaxoft** [comment]
Couple of personal-preference issues I have with the code:
- You're using `cClassName.lua` filenames, so all filenames start with a `c` needlessly.
- There are very few comments. This will make later maintenance quite difficult. You should at least document your classes, what each function does and what parameters it expects, in what format.

File: README.md [hunk 3]
```diff
@@ -1,25 +1,2 @@
...
-   * login.logout
-   * Add login.removeacc to your admin/mod group
- * You have now installed Login for MCServer!
-
-Configuration
--------------
-
-The configuration file for Login is located inside its folder in the Plugins folder. Simple edit `Config.Ini`.
+# Login
+An authentication plugin for MCServer
```
Comment (madmaxoft) [inline README.md]: Why not use the old text, possibly updated, and let it generate from Info.lua?

**Comment by NiLSPACE** [comment]
@madmaxoft 
- true I start files that contain classes with a c. I should probably rename them. How about `class_<classname>` or `c_<classname>`
- Yeah it was pretty dumb of me to not comment while I was writing the code. Although I am pretty happy with the documentation of the SQLiteHandler class. It's basicly a Lua version of a database class I wrote for school.

File: CommandHandlers.lua (hunk 1 above)
Comment (NiLSPACE) [inline CommandHandlers.lua]: Good idea, I'll do that.

**Comment by madmaxoft** [comment]
Why not use just ClassName.lua filename for `cClassName` implementation? Like we do it in the server. Even the `c` prefix for classes is outdated and most modern frameworks don't use it; MCServer uses it only due to compatibility reasons (Lua API)

**Comment by NiLSPACE** [comment]
Really? Didn't know that. I really like the c prefix. It helps alot to differentiate a class and a function.

File: playerstate.lua [hunk 4]
```diff
@@ -0,0 +1,37 @@
+
+
+
+
+
+g_PlayerStates = {}
```
Comment (madmaxoft) [inline playerstate.lua]: Make this `local` - it should not be directly accessed from the outside of this file.

**Pull request merged**